lib/python/picongpu/plugins/energy_histogram.py

```diff
@@ -111,46 +115,68 @@ def get(self, species, species_filter="all", iteration=None,
             (defined in ``particleFilters.param``)
         iteration : (unsigned) int [unitless]
             The iteration at which to read the data.
-            @TODO also allow lists here
+            A list of iterations is allowed as well.
         include_overflow : boolean, default: False
             Include overflow and underflow bins as the first/last bins.
 
         Returns
         -------
         energies : np.array of dtype float [unitless]
             count of particles in each bin
-            @todo if iteration is a list, return a dict
+            If iteration is a list, returns a matrix with first index being
+            the iteration as in the order of the requested iterations.
         bins : np.array of dtype float [keV]
             upper ranges of each energy bin
         """
         if iteration is None:
```

Comment:
Why is it actually that you give a default value to iteration which will throw an exception? 
Since the parameter is needed anyway, wouldn't it be good to not provide a default then?

Reply:
Great question, thanks!
unfortunately, there is no good default for an iteration to select since it is very case dependent.
The whole idea about defaults here is, that we have
- a consequent order of arguments
- allow to select only parts of them to users

That means a user will always see the order "species, filter, iteration, specific_options" in the data and viz API. But if they are visualizing the default "all" filter for example, they can just use the python-specific  syntax to set a "later" optional argument `get("e", iteration=1234)`

It's therefor ok to set all later to default parameters and still require them and common practice in Python APIs.

Reply:
Of course if one is clever one can get it also done with a good order of arguments, but orders sometimes contradict each other for different workflows. Here for example, iteration could always be the first argument, but I would have to check again why I put it last (and if we change it all APIs need the change at once)

Reply:
Alternative approach to your question below: get "all iterations" if `None` (default) is specified

Reply:
Sounds like a plan to hand out all iterations when not specific ones are provided. I like that :)